docs/protocol/network/data-structures/transaction.md

```diff
@@ -0,0 +1,136 @@
+- Commit: used by witnesses to (1) commit the results of their retrieval tasks without revealing the actual retrieval data, and (2) pledge their portion of the witness reward.
+- Reveal: used by witnesses to (1) reveal the actual retrieval data that they commited in their previous *commit*, and (2) pledge their portion of the witness reward.
+
+### P2PKH outputs
+
+P2PKH outputs are used in Value Transfer Transactions (VTT). The `pkh` field is defined as the first 20 bytes of the public key hash.
+
+| Field   |  Type  | Description                         |
+| ------- | :----: | ----------------------------------- |
+| `pkh`   | `[u8]` | Slice of public key hash (20 bytes) |
+| `value` | `u64`  | Transaction value                   |
+
+### Data Request outputs
+
+Data request outputs are used to define how to request the data from Witnet.
+
+| Field              |  Type  | Description                                                                                                                     |
+| ------------------ | :----: | ------------------------------------------------------------------------------------------------------------------------------- |
+| `data_request`     | `[u8]` | Data request script as a byte array                                                                                             |
+| `value`            | `u64`  | Transaction value that will be used as reward to be distributed after consensus has been reached and fees have been substracted |
+| `witnesses`        |  `u8`  | Minimum amount of data request witnesses                                                                                        |
+| `backup_witnesses` |  `u8`  | Number of backup data request witnesses                                                                                         |
+| `commit_fee`       | `u64`  | Miner fee for each valid commit output included in the block during the **commit phase**                                        |
+| `reveal_fee`       | `u64`  | Miner fee for each valid reveal output included in the block during the **reveal phase**                                        |
+| `tally_fee`        | `u64`  | Miner fee for each valid P2PKH output included in the block during the  **tally phase**                                         |
+| `time_lock`        | `u64`  | The UTC Unix timestamp after which data request may be executed                                                                 |
+
+The minimum data request reward to be distributed among nodes that agreeed with the consensus is defined as follows:
+
+```math
+dr_reward_min = value - (witnesses * commit_fee) - (witnesses * reveal_fee) - (witnesses * tally_fee)
+```
+
+During the reveal and tally phase, some committers may not participate in the data request.
+However, the block miner is incentivized to include as many outputs as committers as possible in order to increase the block fees.
```

Comment:
```suggestion
Miners are actually not obliged to include all the reveal transactions and eventually end up assigning rewards to the committers. This is because there is no way for the network to enforce punishment on them for neglecting or trying to conceal those transactions because there's no guarantee that they will be known to them in discrete time or even known whatsoever. 

However, for every of those transactions that they include in a block, they are eligible for collecting special fees as explicitly specified and set aside for them in the original data request output. It is therefore to be expected that miners will include as many of those transactions as known to them as for maximizing their profit.
```